test/networkd-test.py

```diff
@@ -825,35 +838,37 @@ def create_iface(self, ipv6=False, dhcpserver_opts=None):
 mount -t tmpfs none /run/systemd/netif
 [ ! -e /run/dbus ] || mount -t tmpfs none /run/dbus
 # create router/client veth pair
-cat << EOF > /run/systemd/network/test.netdev
+cat <<EOF >/run/systemd/network/test.netdev
 [NetDev]
-Name=%(ifr)s
+Name={ifr}
 Kind=veth
 
 [Peer]
-Name=%(ifc)s
+Name={ifc}
 EOF
 
-cat << EOF > /run/systemd/network/test.network
+cat <<EOF >/run/systemd/network/test.network
 [Match]
-Name=%(ifr)s
+Name={ifr}
 
 [Network]
 Address=192.168.5.1/24
-%(addr6)s
+{addr6}
 DHCPServer=yes
 
 [DHCPServer]
 PoolOffset=10
 PoolSize=50
 DNS=192.168.5.1
-%(dhopts)s
+{dhopts}
 EOF
 
 # run networkd as in systemd-networkd.service
 exec $(systemctl cat systemd-networkd.service | sed -n '/^ExecStart=/ { s/^.*=//; s/^[@+-]//; s/^!*//; p}')
-''' % {'ifr': self.if_router, 'ifc': self.iface, 'addr6': ipv6 and 'Address=2600::1/64' or '',
-       'dhopts': dhcpserver_opts or ''})
+'''.format(ifr=self.if_router,
+           ifc=self.iface,
+           addr6=('Address=2600::1/64' if ipv6 else ''),
+           dhopts=(dhcpserver_opts or ''))
```

Comment:
bionic-* tests are not happy, doesn't like something in this snippet but it's not immediately clear to me what is wrong:

```
networkd-test.py     FAIL non-zero exit status 1
autopkgtest [11:19:29]: test networkd-test.py:  - - - - - - - - - - stderr - - - - - - - - - -
  File "/tmp/autopkgtest.8IlwYs/build.wtY/systemd/test/networkd-test.py", line 873
    os.fchmod(fd, 0o755)
     ^
SyntaxError: invalid syntax
```